### PR TITLE
Fix #672 - Add description, modelerDescription, UID and version UID to bcl results

### LINKED ISSUE #672: BCL Viewer Human Readable Descriptions
Enhancement Request
--------------

It would be really helpful if the Readme files or human readable description and/or human readable description of modeling approach could be displayed when searching the BCL.  As it is, the data displayed is not very descriptive or helpful.  

This is also a problem with the BCL browser online.  Many of the descriptions are cut off; you must download the measure to actually figure out what it is.

## Detailed Description

Screenshot 2024-01-22 091829

### PR DESCRIPTION
**Pull request by jmarrec**
- Fix #672 
- Add description, modelerDescription, UID and version UID to bcl results

image

### CODE REVIEW

File: src/shared_gui_components/Component.cpp [hunk 1]
```diff
@@ -464,157 +481,170 @@ void Component::createAbridgedLayout() {
...
+    {
+      auto* label = new QLabel("Modeler Description");
+      label->setObjectName("H1");
+      mainLayout->addWidget(label);
+    }
+    {
+      auto* content = new QLabel(m_modelerDescription);
+      mainLayout->addWidget(content);
+      content->setWordWrap(true);
+    }
```
Comment (jmarrec) [inline src/shared_gui_components/Component.cpp:525]: Alternative (`#include <QTextEdit>` needed atop)

```suggestion
    {
      auto* content = new QTextEdit(m_modelerDescription);
      content->setStyleSheet("background: #E6E6E6;");
      content->setFixedHeight(70);
      content->setAcceptRichText(false);
      content->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
      content->setTabChangesFocus(true);
      content->setReadOnly(true);

      mainLayout->addWidget(content);
    }
```

image

image

File: src/shared_gui_components/Component.cpp [hunk 2]
```diff
@@ -464,157 +479,172 @@ void Component::createAbridgedLayout() {
...
-  tableWidget->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-  tableWidget->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
-  tableWidget->horizontalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
-  tableWidget->horizontalHeader()->setStretchLastSection(true);
-
-  mainLayout->addWidget(tableWidget);
+  // Description
+  if (!m_description.isEmpty()) {
+    {
+      auto* label = new QLabel("Description");
```
Comment (macumber) [inline src/shared_gui_components/Component.cpp:502]: Might want to show the section even if it's empty?

File: src/shared_gui_components/Component.cpp [hunk 3]
```diff
@@ -464,157 +479,172 @@ void Component::createAbridgedLayout() {
...
+        display.setNum(attribute.valueAsDouble());
+      } else if (type == AttributeValueType::Integer) {
+        display.setNum(attribute.valueAsInteger());
+      } else if (type == AttributeValueType::Unsigned) {
+        display.setNum(attribute.valueAsUnsigned());
+      } else if (type == AttributeValueType::String) {
+        display = attribute.valueAsString().c_str();
+      } else if (type == AttributeValueType::AttributeVector) {
+        AttributeVector attributeVector = attribute.valueAsAttributeVector();
+        // TODO handle this case
```
Comment (macumber) [inline src/shared_gui_components/Component.cpp:594]: Looks like `attribute.toString` might work for the vector case, might be kind of ugly but I think this is rare.

File: src/shared_gui_components/Component.cpp (hunk 1 above)
Comment (macumber) [reply src/shared_gui_components/Component.cpp]: I think we'd need scroll bars here, either for each section like this or one scroll bar for the whole preview.

**Comment by macumber** [comment]
Nice work!

File: src/shared_gui_components/Component.cpp (hunk 2 above)
Comment (jmarrec) [reply src/shared_gui_components/Component.cpp:502]: I think it looks kinda weird when I do that

File: src/shared_gui_components/Component.cpp (hunk 3 above)
Comment (jmarrec) [reply src/shared_gui_components/Component.cpp:594]: It wasn't there before and no one complained, so I didn't change it :)

toJSONString would be an optional, or just managed it by looping.

File: src/shared_gui_components/Component.cpp (hunk 2 above)
Comment (macumber) [reply src/shared_gui_components/Component.cpp:502]: Sounds good

File: src/shared_gui_components/Component.cpp (hunk 3 above)
Comment (macumber) [reply src/shared_gui_components/Component.cpp:594]: Yeah, I kind of doubt anyone is using AttributeVector

**Pull request merged**